contracts/voting-verifier/src/execute.rs

```diff
@@ -13,17 +14,62 @@ use service_registry::msg::QueryMsg;
 use service_registry::state::Worker;
 
 use crate::error::ContractError;
-use crate::events::{EvmMessages, PollEnded, PollStarted, Voted};
+use crate::events::{EvmMessages, PollData, PollEnded, PollStarted, Voted, WorkerSetConfirmation};
 use crate::execute::VerificationStatus::{Pending, Verified};
 use crate::msg::{EndPollResponse, VerifyMessagesResponse};
 use crate::query::is_message_verified;
-use crate::state::{CONFIG, PENDING_MESSAGES, POLLS, POLL_ID, VERIFIED_MESSAGES};
+use crate::state::{
+    PollType, CONFIG, CONFIRMED_WORKER_SETS, PENDING_MESSAGES, PENDING_WORKER_SETS, POLLS, POLL_ID,
+    VERIFIED_MESSAGES,
+};
 
 enum VerificationStatus {
     Verified(Message),
     Pending(Message),
 }
 
+pub fn confirm_worker_set(
+    deps: DepsMut,
+    env: Env,
+    message_id: MessageID,
+    new_operators: Operators,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let snapshot = take_snapshot(deps.as_ref(), &env, &config.source_chain)?;
+    let participants = snapshot.get_participants();
+
+    if CONFIRMED_WORKER_SETS
+        .may_load(deps.storage, new_operators.hash())?
+        .is_some()
+    {
+        return Err(ContractError::WorkerSetAlreadyConfirmed {});
+    }
+
+    let id = create_poll(
+        deps.storage,
+        env.block.height,
+        config.block_expiry,
+        snapshot,
+        1,
+        PollType::ConfirmWorkerSet,
+    )?;
```

Comment:
have multiple `create_X_poll` instead? Then we can absorb this weird `1`, since the worker set poll always has a single element